### ISSUE TITLE
Set Downloadfolder to mounted NTFS drive? (Linux/Manjaro)

### ISSUE THREAD

**Issue by CaptainCoward** [weechat/weechat#2010]
I now tried all kind of options to set a different download folder on a NTFS drive of mine Using ` /set xfer.file.download_path <path to folder> .` However if doing so the Download just refuses to start. It works with Hexchat, or using a different Folder on my additional ext4 drive. So i figure it is related to the NTFS drives being mounted as root (can you even mount them as User?).
Any Idea? How i can get weechat to use my NTFS drive ?

**Comment by flashcode** [comment]
WeeChat just tries to write a regular file when a file is received via DCC.

Did you check that your user (running WeeChat) has the permissions to writes files in this directory?

**Comment by CaptainCoward** [comment]
Yup i have tried now pretty much anything. And don't seem to have this issue with any other Tool. For example creating files with Double Commander or using Hexchat for  receiving IRC Files just works fine .
So i don't know why weechat would be different here ?
I used to mount my drives via systemd units (.mount files). Though no matter how i mounted it that way it just doesn't seem to work. Which is odd since my ext4 drive is mounted with the same options and is working just fine.
So i tried mounting the ntfs drive with fstab instead (since i saw it was mounted by root unlike my ext4 drive)  that way i got it working by setting the following options within fstab `noatime,uid=1000,gid=1001,rw,user,exec,umask=000 0 0`

though i have no clue why that  works while using a .mount file with pretty much the same settings  
`Options=defaults,rw,noatime,uid=1000,gid=1001` doesn't?  When checking the file permissions with stat i get 
`Access: (0777/drwxrwxrwx)  Uid: (    1000/    Captain)   Gid: (    1001/    Captain)`  which is the same that i would get when mounting with fstab (which works for whatever reason) .  So i don't think it should be related to permissions ?

**Comment by flashcode** [comment]
Is there an error displayed on core buffer (buffer with number 1) when the file fails to be created?

**Comment by CaptainCoward** [comment]
Nope nothing. It just instantly fails to transfer the file. 
I kinda gave up on it now. I can use that drive with Steam (and games just download fine on it), i can create a file (by touch) on that drive without issues and i double checked the permissions (0777/drwxrwxrwx) and yet it just refuses to work  for downloads with weechat when using systemd units instead of fstab for automounting for some reason lol. I now just switched for Hexchat since it works without issues there. Though i'm still curious about a solution to this. Maybe i'm doing something wrong or missed something.

**Comment by flashcode** [comment]
You should see a message in the core buffer whenever a transfer fails (whatever the reason).
What's the message displayed there?
With the message, it will help to locate the problem in the code and maybe by adding some debug information we can find why the download fails.

**Comment by CaptainCoward** [comment]
By "no message" i meant the only message i get is "can't create local file"

**Comment by flashcode** [comment]
Hi,

I just pushed a commit to display a message when the file can not be created, that includes the system error; so you should see the exact reason why the file creation failed.

Output is something like this on core buffer:

```
=!= │ xfer: unable to write file "/path/to/file.txt.part": No such file or directory
```

(in this case the path `/path/to` doesn't exist)

Could you try to compile the master branch and paste the message displayed in your case?

Thanks.

**Comment by flashcode** [comment]
@CaptainCoward: please let me know if you can test this, then we can continue the investigation on your issue, and fix WeeChat if ever there's a bug.
